craft_parts/executor/executor.py

```diff
@@ -280,7 +280,7 @@ def __enter__(self) -> "ExecutionContext":
         self._executor.prologue()
         return self
 
-    def __exit__(self, *exc):
+    def __exit__(self, *exc: Any) -> None:
```

Comment:
Do we want to register this class as a subclass of `contextlib.AbstractContextManager`?

Reply:
I don't really have a strong reasoning either way. I'm just in a habit of doing stuff like that because it's saved me lots of debugging with IDE tools in the past. In this case I meant the question literally and didn't mean to imply it as a suggestion.

Reply:
oh alright! I could have done that but I merged the PR after this was marked as resolved. Oops!